wasm_validator/derive/src/lib.rs

```diff
@@ -0,0 +1,228 @@
+//! Crate with validator-related derive macros.
+
+#![allow(clippy::panic)]
+
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, parse_quote, DeriveInput};
+
+mod conversion;
+mod entrypoint;
+mod token;
+mod validate;
+
+/// Annotate the user-defined function that starts the execution of a validator.
+///
+/// Validators are only checking if an operation is **invalid**, not if it is valid.
+/// A validator can either deny the operation or pass it to the next validator if there is one.
+///
+/// # Attributes
+///
+/// This macro must have an attribute describing entrypoint parameters.
+///
+/// The syntax is:
+/// `#[iroha_wasm::validator_entrypoint(params = "[<type>,*]")]`, where `<type>` is one of:
+/// - `authority` is a signer account id who submits an operation
```

Comment:
```suggestion
/// The syntax is:
/// 
/// ```ignore
/// #[iroha_wasm::validator_entrypoint(params = "[<type>,*]")]
/// ```
///
/// where `<type>` is one of:
///
/// - `authority` is a signer account id who submits an operation
```